### ISSUE TITLE
Flaky test: TestRetryStreaming

### ISSUE THREAD

**Issue by menghanl** [grpc/grpc-go#2188]
```
--- FAIL: TestRetryStreaming (1.07s)
	retry_test.go:542: Multiple successive requests: client: Send(payload:<body:"\002" > ) = EOF; want <nil>
	retry_test.go:547: Multiple successive requests: serverOpIter = 2; want 9
```

**Comment by dfawley** [comment]
I'm able to reproduce this now, and can isolate it to a single run of this test case (i.e. there is no state leaking between test functions or test cases).  It takes many iterations to reproduce, but it should be enough to debug.

**Comment by dfawley** [comment]
Fixed by #2199
